emily_cron/test/test_deposit_processor.py

```diff
@@ -365,43 +366,191 @@ def tearDownModule():
+            txid="replacement2",
+            confirmed_height=700000,
+            lock_time=10,
+            rbf_txids=["replacement1", "replacement2"],
+        )
+
+        # Mock the transaction data returned by the Mempool API
+        tx1_data = {
+            "vin": [{"prevout": {"value": 2000000}}],
+            "vout": [{"scriptpubkey_address": "bc1q...", "value": 1900000}],
+            "fee": 100000,
+            "status": {"block_height": None, "block_time": None},  # Unconfirmed
+        }
+
+        tx2_data = {
+            "vin": [{"prevout": {"value": 2000000}}],
+            "vout": [{"scriptpubkey_address": "bc1q...", "value": 1900000}],
+            "fee": 100000,
+            "status": {"block_height": 700000, "block_time": self.current_time - 3600},  # Confirmed
         }
 
+        mock_get_tx.side_effect = lambda txid: tx1_data if txid == "replacement1" else tx2_data
+
+        # Mock the RBF check
+        mock_check_rbf.side_effect = lambda txid: (
+            {"replacement1", "replacement2"} if txid == "replacement1" else set()
+        )
+
+        # Mock the from_deposit_info method, because MagicMock would not work with asdict
+        with patch("app.models.EnrichedDepositInfo.from_deposit_info") as mock_from_info:
+            # Set up the mock returns
+            def mock_from_info_side_effect(deposit, additional_info):
+                enriched = MagicMock(spec=EnrichedDepositInfo)
+                enriched.bitcoin_txid = deposit.bitcoin_txid
+                enriched.confirmed_height = additional_info.get("confirmed_height")
+                enriched.confirmed_time = additional_info.get("confirmed_time")
+                enriched.fee = additional_info.get("fee")
+                enriched.in_mempool = additional_info.get("in_mempool")
+                enriched.rbf_txids = additional_info.get("rbf_txids")
+                return enriched
+
+            mock_from_info.side_effect = mock_from_info_side_effect
+
+            # Run the _enrich_deposits method
+            result = self.processor._enrich_deposits([deposit1, deposit2])
+            # Verify the correct methods were called
+            mock_get_tx.assert_any_call("replacement1")
+            mock_get_tx.assert_any_call("replacement2")
+
+            # Verify RBF check was called only for the unconfirmed transaction
+            mock_check_rbf.assert_called_once_with("replacement1")
+
+            # Verify from_deposit_info was called with the correct parameters
+            self.assertEqual(mock_from_info.call_count, 2)
+
+            # Assertions on the configured mock results
+            self.assertEqual(result[0].bitcoin_txid, deposit1.bitcoin_txid)
+            self.assertEqual(result[0].confirmed_height, deposit1.confirmed_height)
+            self.assertEqual(result[0].fee, 100000)
+            self.assertEqual(result[0].in_mempool, True)
+            self.assertEqual(result[0].rbf_txids, {"replacement1", "replacement2"})
```

Comment:
nit: can we add also the same check for `result[1]`, checking that `rbf_txids` is empty?